dex-it/src/test/scala/com/wavesplatform/it/sync/MarketOrderTestSuite.scala

```diff
@@ -1,118 +1,449 @@
 package com.wavesplatform.it.sync
 
+import java.nio.charset.StandardCharsets
+
 import com.typesafe.config.{Config, ConfigFactory}
 import com.wavesplatform.account.KeyPair
+import com.wavesplatform.common.state.ByteStr
 import com.wavesplatform.dex.model.MatcherModel.Normalization
 import com.wavesplatform.it.MatcherSuiteBase
+import com.wavesplatform.it.api.MatcherStatusResponse
 import com.wavesplatform.it.api.SyncHttpApi._
 import com.wavesplatform.it.api.SyncMatcherHttpApi._
-import com.wavesplatform.it.api.{LevelResponse, MatcherStatusResponse}
 import com.wavesplatform.it.sync.config.MatcherPriceAssetConfig._
+import com.wavesplatform.transaction.Asset.IssuedAsset
 import com.wavesplatform.transaction.assets.exchange.OrderType.{BUY, SELL}
 import com.wavesplatform.transaction.assets.exchange.{AssetPair, OrderType}
-import mouse.any._
 
 class MarketOrderTestSuite extends MatcherSuiteBase {
+  val fixedFee   = 0.003.waves
+  val percentFee = 14
 
   implicit class DoubleOps(value: Double) {
     val waves, eth: Long = Normalization.normalizeAmountAndFee(value, 8)
     val usd: Long        = Normalization.normalizePrice(value, 8, 2)
   }
 
+  def tooLowPrice(orderType: String, price: String): String = {
+    s"Price of the $orderType market order ($price) is too low for its full execution with the current market state"
+  }
+
+  def tooHighPrice(orderType: String, price: String): String = {
+    s"Price of the $orderType market order ($price) is too high for its full execution with the current market state"
+  }
+
   override protected def nodeConfigs: Seq[Config] = {
-    super.nodeConfigs.map(ConfigFactory.parseString("waves.dex.allowed-order-versions = [1, 2, 3]").withFallback)
+    val orderFeeSettingsStr =
+      s"""
+         |waves.dex {
+         |  allowed-order-versions = [1, 2, 3]
+         |  order-fee {
+         |    mode = percent
+         |    dynamic {
+         |      base-fee = 300000
+         |    }
+         |    percent {
+         |      asset-type = amount
```

Comment:
In your test suite you don't take into account another asset types in percent mode
Maybe it makes sense to add tests for this sub-mode too